frontend/src/metabase/query_builder/components/view/sidebars/QuestionInfoSidebar/components/QuestionSources.tsx

```diff
@@ -0,0 +1,64 @@
+import { Fragment, useMemo } from "react";
+import { match } from "ts-pattern";
+import { c } from "ttag";
+
+import { SidesheetCardSection } from "metabase/common/components/Sidesheet";
+import Link from "metabase/core/components/Link";
+import { getIcon } from "metabase/lib/icon";
+import { Box, Flex, FixedSizeIcon as Icon, type IconProps } from "metabase/ui";
+import type Question from "metabase-lib/v1/Question";
+
+import { getDataSourceParts } from "../../../ViewHeader/components/QuestionDataSource/utils";
+
+interface QuestionSource {
+  href: string;
+  name: string;
+  model?: string;
+  iconProps: Omit<IconProps, "display">;
+}
+
+export const QuestionSources = ({ question }: { question: Question }) => {
+  const sources = getDataSourceParts({
+    question,
+    subHead: false,
+    isObjectDetail: true,
+    formatTableAsComponent: false,
+  }) as unknown as QuestionSource[];
+
+  const sourcesWithIcons: QuestionSource[] = useMemo(() => {
+    return sources.map(source => {
+      const modelForIcon = match(source.model)
+        .with("question", () => ({ model: "card" as const }))
+        .with("model", () => ({ model: "dataset" as const }))
+        .otherwise(() => ({ model: "table" as const }));
```

Comment:
we should have a database icon next to databases,



![Screen Shot 2024-10-08 at 9 41 48 AM](https://github.com/user-attachments/assets/5a11f210-a51a-4330-9ac6-243d35f8ca1f)

also, for some reason the model/question distinction is not working. the screenshot below should have a model icon, which is properly shown in the header

![Screen Shot 2024-10-08 at 9 44 07 AM](https://github.com/user-attachments/assets/27e74ece-80ef-49e2-9efe-6c2302858af7)

I suppose having schema in here is nice, though I don't recall seeing it specced, but i'm not sure what Icon they should have?

![Screen Shot 2024-10-08 at 9 45 27 AM](https://github.com/user-attachments/assets/547e5aff-3ea4-4947-b548-ec6b14112412)